script.js

```diff
@@ -53,81 +79,247 @@ my_widget_script =
             }
             return true;
           }
-         
+
           $('#the_form tbody tr').each(function() {
-              if (isEmpty(this)) {
-                $(this).remove();
-              }
-            })
+            if (isEmpty(this)) {
+              $(this).remove();
+            }
+          })
           return;
         }
-       
+
         var nFixed = 2;
-        $('#the_form input[name^=amount]').on('keyup change', function() {
+
+        /*
+			Amount Field On Blur Handler
+		*/
+        $('#the_form input[name^=amount]').on('blur', function() {
           var tr = $(this).closest('tr');
-          var equiv = $('input[name^=equivalents]', tr);
-          if (equiv.length != 1)
-            throw ('Found the incorrect number of equiv inputs: ' + equiv.length)
-          equiv = equiv.first();
-          var moles1 = $('#the_form input[name=moles1_formula]');
+          console.log('In amount blur handler'+ tr.attr('class'))
+
+          current_class = tr.attr('class')
+          var compare_class_id = current_class.localeCompare('initialRow');
           var fw = $('input[name^=fw]', tr);
-          if ($(this).val() && fw.val() && moles1.val())
-            equiv.val(($(this).val() / fw.val() / moles1.val()).toFixed(nFixed))
+          var amount = $('input[name^=amount]', tr);
+          var moles = $('input[name^=moles]', tr);
+          if(fw.val() && amount.val()) {
+            var moles_1 = amount.val() / fw.val()
+            console.log(moles_1)
+            if (!moles.val()) {
+              moles.val(moles_1.toFixed(nFixed))
+            }
+          }
+
+          if (moles.val() && fw.val() && amount.val && (compare_class_id == 0))  {
+            my_widget_script.enable_all_records()
+          }
+
+          // for rows greater than 1
+          if ( (amount.val()) && (fw.val()) && (compare_class_id != 0)) {
```

Comment:
* don't put a space after an opening parenthesis unless you put a space before the corresponding closing parenthesis
* don't place redundant parentheses around expressions when order is already maintained by [operator precedence](https://www.w3schools.com/js/js_arithmetic.asp)
* don't compare to a bool with `!=` and `==`

```suggestion
          if (amount.val() && fw.val() && compare_class_id) {
```